falcon_sqla/manager.py

```diff
@@ -122,6 +136,29 @@ def get_session(self, req=None, resp=None):
 
         return self._Session()
 
+    def close_session(self, session, succeeded, req=None, resp=None):
+        """Close a session obtained via :func:`get_session`.
+
+        .. note:: There is no need to invoke this method manually if you are
+                  using the :func:`session_scope` context manager, or if you
+                  are using middleware.
+        """
+        session_cleanup = self.session_options.session_cleanup
+
+        try:
+            if (session_cleanup == COMMIT_ON_SUCCESS and succeeded
+                    or session_cleanup == COMMIT):
+                session.commit()
+            elif session_cleanup != CLOSE_ONLY:
+                session.rollback()
+        finally:
+            if req and resp:
+                # NOTE(vytas): Break circular references between the request
+                #   and session in case the latter was stored in req.context.
+                del session.info['req']
```

Comment:
not sure if it makes sense to also do `req.context.session = None`

Reply:
Dunno, it's hopefully not needed really, as long as we delete all the references from the session's context, there should be no circular dependencies there (Python can handle these too, but it usually takes more time).